cpp/oneapi/dal/graph/test/service_functions_test.cpp

```diff
@@ -0,0 +1,187 @@
+#include "oneapi/dal/test/engine/common.hpp"
+
+#include <fstream>
+#include <array>
+
+#include "oneapi/dal/graph/service_functions.hpp"
+#include "oneapi/dal/io/load_graph.hpp"
+
+namespace oneapi::dal::graph::test {
+
+  class graph_base_data {
+  public:
+    graph_base_data() = default;
+
+    void write_test_data() {
+      std::ofstream outf(filename);
+      if (outf.is_open()) {
+        outf << file_content;
+      } else {
+        throw invalid_argument(dal::detail::error_messages::file_not_found());
+      }
+      outf.close();
+    }
+
+    void delete_test_data() { std::remove(filename.c_str()); }
+
+    const std::string get_filename() const { return filename; }
+
+    std::int64_t get_correct_vertex_count() const { return vertex_count; }
+
+    std::int64_t get_correct_edge_count() const { return edge_count; }
+
+  protected:
+    std::string filename;
+    std::string file_content;
+    std::int64_t vertex_count;
+    std::int64_t edge_count;
+  };
+
+  class two_vertices_graph_type : public graph_base_data {
+  public:
+    two_vertices_graph_type() {
+      filename = "test_two_vertices_graph.csv";

```

Comment:
@rlnx Is it ok that we write file in current folder? No absolute path as it anyway will be removed. File names for each type of graph are different

Reply:
It's ok if we write to current folder, but it should be separate tests for IO, but not tests for service functions.